src/frontend/src/optimizer/plan_node/generic/sys_scan.rs

```diff
@@ -0,0 +1,286 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::rc::Rc;
+
+use educe::Educe;
+use pretty_xmlish::Pretty;
+use risingwave_common::catalog::{ColumnDesc, Field, Schema, TableDesc};
+use risingwave_common::util::column_index_mapping::ColIndexMapping;
+use risingwave_common::util::sort_util::ColumnOrder;
+
+use super::GenericPlanNode;
+use crate::catalog::ColumnId;
+use crate::expr::ExprRewriter;
+use crate::optimizer::optimizer_context::OptimizerContextRef;
+use crate::optimizer::property::{Cardinality, FunctionalDependencySet, Order};
+use crate::utils::{ColIndexMappingRewriteExt, Condition};
+
+/// [`SysScan`] returns contents of a table or other equivalent object
+#[derive(Debug, Clone, Educe)]
+#[educe(PartialEq, Eq, Hash)]
+pub struct SysScan {
+    pub table_name: String,
```

Comment:
May be `enum` in the future.

Reply:
Table name as an enum?

Reply:
Or do you mean the core `SysScan` can be part of some enum 🤔

Reply:
> Table name as an enum?

Yeah, I mean this, because system tables are fully controlled by us, so there can be an `enum` containing all available system tables. 

Not a big deal.